beacon_node/eth2-libp2p/src/behaviour/handler/mod.rs

```diff
@@ -77,15 +89,25 @@ impl<TSpec: EthSpec> ProtocolsHandler for BehaviourHandler<TSpec> {
         out: <Self::OutboundProtocol as OutboundUpgrade<NegotiatedSubstream>>::Output,
         info: Self::OutboundOpenInfo,
     ) {
+        if self.keep_alive < KeepAlive::Yes {
+            // shutting down, reject all outbounds
+            return;
+        }
         self.delegate.inject_fully_negotiated_outbound(out, info)
     }
 
     fn inject_event(&mut self, event: Self::InEvent) {
+        if self.keep_alive < KeepAlive::Yes {
+            // reject everything! TODO: what can keep going?
```

Comment:
Yeah, lets reject everything here. 
Its important to notify the error for the RPC. So I'm not sure where is the best place for doing this. In the RPC or here. If we do it in the RPC, we should allow these to the RPC Handler which will then send back the error.